src/Symfony/Component/Serializer/Normalizer/ObjectNormalizer.php

```diff
@@ -194,7 +194,12 @@ protected function isAllowedAttribute($classOrObject, string $attribute, ?string
         $class = \is_object($classOrObject) ? \get_class($classOrObject) : $classOrObject;
 
         if ($context['_read_attributes'] ?? true) {
-            return $this->propertyInfoExtractor->isReadable($class, $attribute) || $this->hasAttributeAccessorMethod($class, $attribute);
+            return (\is_object($classOrObject) && $this->propertyAccessor->isReadable($classOrObject, $attribute))
+                || $this->propertyInfoExtractor->isReadable($class, $attribute) || $this->hasAttributeAccessorMethod($class, $attribute);
+        }
+
+        if (str_contains($attribute, '.')) {
```

Comment:
I don't know a better way to fix this. The `$this->propertyAccessor->isWritable()` method expects an object, but the `isAllowedAttribute()` method is called before the object is instantiated.

Perhaps the `$this->propertyAccessor->isReadable()` check should be removed as well, and the method should simply return `true` if the attribute contains a `.`?

Reply:
Indeed, I think that when can only rely on the `.` in the first place, as it'll stick a bit more with the legacy behavior.

But later on I do think that it can be great to be able to know if `foo.bar.baz` is readable/writable only based on reflection, so we have a more accurate behavior.